ribbon-loadbalancer/src/main/java/com/netflix/loadbalancer/AbstractServerPredicate.java

```diff
@@ -137,6 +137,22 @@ public void setLoadBalancerStats(LoadBalancerStats stats) {
             return results;            
         }
     }
+	
+	/**
+	 * Referenced from RoundRobinRule
+     * Inspired by the implementation of {@link AtomicInteger#incrementAndGet()}.
+     *
+     * @param modulo The modulo to bound the value of the counter.
+     * @return The next value.
+     */
+    private int incrementAndGetModulo(int modulo) {
```

Comment:
Hi, the potential busy spin here is a little concerning. What about something like
```
int current = Math.abs(nextIndex.incrementAndGet());
return current % modulo;
```

Reply:
as its comment suggests, it's the same as RoundRobinRule.incrementAndGetModulo(int modulo)(in master branch). it should be safe. 

and for "int current = Math.abs(nextIndex.incrementAndGet()); "

java docs writes "Note that if the argument is equal to the value of Integer.MIN_VALUE, the most negative representable int value, the result is that same value, which is negative."
https://docs.oracle.com/javase/7/docs/api/java/lang/Math.html#abs(int)

it seems that could be an issue when it's overflowed. 

@pacoyi is the one who made the PR though. :-)

Reply:
Fair point, that would indeed be an issue.